File: Programmer_test/PT_Python/P120860.py
def solution(dots):
    ma_x, ma_y, mi_x, mi_y = dots[0][0], dots[0][1], dots[0][0], dots[0][1]
    for i in range(1,4):
        if (ma_x < dots[i][0]):
            ma_x = dots[i][0]
        if (mi_x > dots[i][0]):
            mi_x = dots[i][0]
        if (ma_y < dots[i][1]):
            ma_y = dots[i][1]
        if (mi_y > dots[i][1]):
            mi_y = dots[i][1]
    return (ma_x - mi_x) * (ma_y - mi_y)

print(solution([[1, 1], [2, 1], [2, 2], [1, 2]]))
print(solution([[-1, -1], [1, 1], [1, -1], [-1, 1]]))

File: Programmer_test/PT_Java/P120860.java
public class P120860 {
    public static void main(String[] args) {
        P120860 T = new P120860();
        System.out.println(T.solution(new int[][] {{1,1}, {2,1}, {2,2}, {1,2}}));
        System.out.println(T.solution(new int[][] {{-1,-1}, {1,1}, {1,-1}, {-1,1}}));
    }  

    public int solution(int[][] dots) {
        int ma_x = dots[0][0];
        int mi_x = dots[0][0];
        int ma_y = dots[0][1];
        int mi_y = dots[0][1];
    
        for (int i = 1; i < 4; i++) {
            if (ma_x < dots[i][0]) 
                ma_x = dots[i][0];
            if (mi_x > dots[i][0]) 
                mi_x = dots[i][0];
            if (ma_y < dots[i][1]) 
                ma_y = dots[i][1];
            if (mi_y > dots[i][1]) 
                mi_y = dots[i][1];
        }
        return (ma_x - mi_x) * (ma_y - mi_y);
    }
}